dip-0001.md

```diff
@@ -0,0 +1,190 @@
+
+## Consensus Protocol Changes
+
+Currently there is a protocol rule as follows:
+* A block over 1MB is invalid. 
+
+To allow the network to scale we replace this rule with the two rules:
+* A block over 2MB is invalid.
+* A block with at least one transaction over 100kB is invalid.
+
+Note the 100kB cap on transaction size will not necessarily be satisfied by old blocks. For this reason this new rule must only be enforced on blocks after activation of the protocol change. 
+
+## Observation
+
+### 100kB  Transaction Limit
+
+This subsection is to only point out a possible efficiency of the 100kB transaction size limit. Currently, transactions in the mempool already verified to be standard and thus satisfy the size limit.  It should then be possible that only transactions not seen by the node would need to be checked to satisfy the 100kB limit.  However, checking if a transaction is in the mempool might take as long as verifying the size.  
+
+## BIP 009 Inspired Activation
+
+To ensure a coherent network roll-out we employ BIP9 signaling adjusted for Dash.  We reserve the last bit in the version to signal for DIP1 activation.  A DIP0001 enabled miner will broadcast block version `0x20000001` only if the masternode selected for that block is also DIP1 enabled.  We model the 2016 block difficulty adjustment period in BIP9 with a 4032 block round.  This length of a round should allow for a representative sample of masternodes to be taken.  
+ 
+As a block is mined the version number is determined by the miner who mined it and the masternode that is selected for that block.   Note that blocks mined by non upgraded miners will broadcast version `0x20000000` without regard for the DIP1 acceptance of the masternode selected.
+ 
+As in BIP9 each block is assigned a state according to the flow chart:
+
+![State Flow Chart](/dip-0001/states.png)
+
+Above, MTP represents the median time past as in BIP113. The variables starttime and timeout are particular integers that represent the time in the current epoch.  Here, threshold reached, is a particular boolean we will define below, and denote without the space,  thresholdreached, in code.
+
+The state of every block is determined by the algorithm below.
+
+The state of the Genesis block is DEFINED as in BIP009
+
+```C++
+State GetStateForBlock(block) {
+  if (block.height == 0) { 
+    return DEFINED;
+  }
+```
+
+All blocks in the same round have the same state.
+
+```C++
+if ((block.height % 4032) != 0) {
+            return GetStateForBlock(block.parent);
+        }
+```
+
+Otherwise, the next state depends on the previous state:
+
+```C++
+switch (GetStateForBlock(GetAncestorAtHeight(block, block.height - 4032))) {
+```
+
+We remain in the DEFINED state until either we pass the starttime or the timeout. GetMedianTimePast in the code below refers to the median nTime of a block and its 10 predecessors. 
```

Comment:
We remain in the DEFINED state until we either pass the starttime or the timeout.